### PR TITLE
Docker support + Fix issue on road tiles

### PR DESCRIPTION
**Pull request by tiagojsag**
- Add docker support to the cruncher (see https://github.com/GFDRR/osm-analytics-cruncher/issues/6 and https://github.com/GFDRR/osm-analytics-cruncher/issues/14)
- Fix issue with road tiles not being generated (see https://github.com/hotosm/osm-analytics/issues/63)

### CODE REVIEW

File: app/crunch.sh [hunk 1]
```diff
@@ -13,15 +13,15 @@ mkdir -p intermediate
 
 # apply filter, merge with user experience data, aggregate to bins
 # and create z13-z14 tiles for raw data
-./src/index.js $1 $2.json
+node ./src/index.js $1 $2.json
```
Comment (tyrasd) [inline app/crunch.sh:16]: Any particular reason for this change? ~~On some (debian) systems, node's binary is not called node, but *nodejs* by default. That's why I used `#!/usr/bin/env node` on executables, like here: https://github.com/GFDRR/osm-analytics-cruncher/blob/ac9fcacb826d8a1483c206fca5783128b416e3e6/app/src/index.js#L1~~ (edit: after thinking about it a second time, I guess that having the `env node` thing in the hashbang doesn't make a difference regarding the debian nodejs naming issue, so it really doesn't matter I guess)

**Comment by tyrasd** [comment]
> fix issue on road tiles

There's still one more issue with the road data, unfortunately: The higher zoom levels (zoomed all in where you actually see the linestrings and the first level of aggregation) appear to be fine, but the lower zooms generate incorrect results. One sees this quite clearly in the history tab:

selection_115

(This should match the numbers I generated for my SotM talk: http://bl.ocks.org/tyrasd/raw/1895193efb9fa49c776dee73ab5d717c/#17 – current total length of all OSM *highway* lines in Italy should be well over 1,000,000 km.)

This bug is likely related to the fact that the first aggregation zoom-level is twice as fine-grained as the rest of the zoom levels. I believe this was unintentional, right? It looks like https://github.com/GFDRR/osm-analytics-cruncher/blob/1096f92a3c8986da7a2cb3da88966d28e02b16ca/app/crunch.sh doesn't pass the `BINNINGFACTOR` onto the `index.js` script (which looks like a bug [I introduced](https://github.com/hotosm/osm-analytics-cruncher/commit/5b2be0318a4b94dc1f8434c8f79a6ba2dc10238f#diff-1db3fedf2b74b478edb83c59f436c032) when working on the notippecanoe branch – sorry :fearful:).

Luckily this should be a quick fix. Also, since the current "cruncher" seems to be able to handle the binning factor of 64 for roads just fine, why not bump it up to 64 in https://github.com/GFDRR/osm-analytics-cruncher/blob/1096f92a3c8986da7a2cb3da88966d28e02b16ca/app/run.sh#L52 ? (Originally, we did actually want to have the roads layer as fine grained as the building layer, but some technical issues made the cruncher constantly crash if the binning factor wasn't set to 32 for the roads – I don't remember exactly what it was, but it could be likely something to do with tippecanoe being in the pipeline.)

**Comment by tyrasd** [comment]
> Docker support

Looks like a great addition for the project, but I cannot really review that part of the PR, since I don't actually know very much about docker.

**Comment by tiagojsag** [comment]
hi @tyrasd 

Thanks for the feedback. If I understand correctly, the only action necessary here would be the road data issue fix. My colleague @rrequero who developed the current fix, will be back from holiday next week, so I'll go over the issue with him. But, if it indeed a simple fix, as you suspect, we should be able to do it as part of this PR.

Thanks

**Comment by rrequero** [comment]
Hi @tyrasd and @tiagojsag 
I changed the factor of highways generation. Now the size of the result file is 32GB (The size is the double). Also, the machine takes a 1 hour more to generate the result files.

Thanks
Raúl

**Comment by tyrasd** [comment]
finally this is merged. :bowing_man: I fixed the one remaining bug outlined above in 6d5fd29.

**Pull request closed without merging**